src/compressor.h

```diff
@@ -66,7 +74,13 @@ struct ScriptCompression
         unsigned int nSize = 0;
         s >> VARINT(nSize);
         if (nSize < nSpecialScripts) {
-            std::vector<unsigned char> vch(GetSpecialScriptSize(nSize), 0x00);
+            auto size = GetSpecialScriptSize(nSize);
+            // if size > sizeof(CompressedScript) this would incur a
+            // heap allocation in the CompressedScript prevector.
+            // bump the CompressedScript prevector size if this ever
+            // happens.
+            assert(size <= sizeof(CompressedScript));
```

Comment:
This line is a little frightening to me.

Is this something we know at compile time and can static_assert? Otherwise we should just resize and take the performance hit here.

Performance degradation is much better than crash fail :)

Reply:
I was hoping this assert would trigger in the tests if `GetSpecialScriptSize` was ever changed, but I just tried chaging the 32 value to 34 and the assert wasn't triggered. so now I'm wondering if there should be better test coverage here.

Reply:
oh nm I just realized this is horribly wrong, not sure why I'm doing sizeof here. I'm going to just delete this assert